shell/src/main/java/org/apache/accumulo/shell/commands/QuotedStringTokenizer.java

```diff
@@ -109,7 +110,18 @@ private void createTokens() throws BadArgumentException, UnsupportedEncodingExce
         } else {
           token[tokenLength++] = inputBytes[i];
         }
+      } else if (inValue) {
+        if (ch != ' ') {
+          token[tokenLength++] = inputBytes[i];
+        } else {
+          inValue = false;
+          tokens.add(new String(token, 0, tokenLength, Shell.CHARSET));
+          tokenLength = 0;
+        }
       } else {
+        if (ch == '=') {
+          inValue = true;
+        }
```

Comment:
Seems like the rest of the if checks could be chained.  Seems like they are all looking for mutually exclusive chars.

```suggestion
        } else if (ch == '\'' || ch == '"')  {
```

Reply:
Confirmed that the check chaining worked and then refactored to use switch statements when doing constant comparisons. This change made the parsing logic easier to follow.